solver/src/solver/paraswap_solver.rs

```diff
@@ -57,40 +60,132 @@ impl ParaswapSolver {
     }
 }
 
+fn is_retryable_err(err: &SettlementError) -> bool {
+    match err {
+        SettlementError::Paraswap(ParaswapResponseError::PriceChange) => {
+            tracing::warn!("Paraswap Price Change");
+            true
+        }
+        SettlementError::Paraswap(ParaswapResponseError::BuildingTransaction(query)) => {
+            tracing::warn!("Paraswap ERROR_BUILDING_TRANSACTION from query {}", query);
+            true
+        }
+        _ => false,
+    }
+}
+
+enum SettlementError {
+    Anyhow(anyhow::Error),
+    Paraswap(ParaswapResponseError),
+}
+
+impl From<anyhow::Error> for SettlementError {
+    fn from(err: Error) -> Self {
+        SettlementError::Anyhow(err)
+    }
+}
+
+impl From<ParaswapResponseError> for SettlementError {
+    fn from(err: ParaswapResponseError) -> Self {
+        SettlementError::Paraswap(err)
+    }
+}
+
+impl From<SettlementError> for anyhow::Error {
+    fn from(err: SettlementError) -> Self {
+        match err {
+            SettlementError::Paraswap(err) => anyhow!("Paraswap Response Error {:?}", err),
+            SettlementError::Anyhow(err) => err,
+        }
+    }
+}
+
 #[async_trait::async_trait]
 impl SingleOrderSolving for ParaswapSolver {
     async fn settle_order(&self, order: LimitOrder) -> Result<Option<Settlement>> {
-        let (amount, side) = match order.kind {
-            model::order::OrderKind::Buy => (order.buy_amount, Side::Buy),
-            model::order::OrderKind::Sell => (order.sell_amount, Side::Sell),
-        };
-        let token_infos = self
+        let max_retries = 2;
+        for _ in 0..=max_retries {
+            match self.try_settle_order(order.clone()).await {
+                Ok(settlement) => return Ok(settlement),
+                Err(err) if is_retryable_err(&err) => continue,
```

Comment:
I'd print something like:
```rust
tracing::debug!("Retrying due to: {:?}", err)
```
 here instead of in the `is_retryable_error` method